### PR TITLE
Turrets can shoot blobs

### LINKED ISSUE #31456: [FEATURE REQUEST] AI Turrets/ Any Turrets should shoot at Blob
(WEB REPORT BY: primed REMOTE: 206.221.180.138:7777)
# Revision
e9cc5567301931778ecb32597f261757b2d3c03c

# Description
watching the blob slowly creep up on the AI with nothing he can do is very odd to me

### PR DESCRIPTION
**Pull request by gurfan**
Closes: #31456
Gun Turrets + Portable (Buildable) Turrets will shoot blob if they have "shoot unidentified lifesigns" on.
AI turrets will shoot blob if they are turned on.

:cl:
 * tweak: All types of turrets can now target blobs.

### CODE REVIEW

**Comment by Kromkar** [comment]
Blob will never win again if people can shit out infinite ammo turrets

**Comment by boy2mantwicethefam** [comment]
@Kromkar Just buff blob, duh

**Comment by gurfan** [comment]
Blob doesn't need a buff. I've tested this and the turrets aren't as powerful against blobs as you think.

**Comment by Kurfursten** [comment]
The rate turrets fire at makes this much less threatening to blob than emitters. Turrets don't need to be directly wired, but they're also immobile.

I am inclined to believe Gurf in that his testing turned up turrets being not much help against blob, but I will give this at least 3 days for people to vote and check out the branch if they want. What I can say is that even with liberators blob is still performing very well in direct firefight situations so I don't expect this to impede that.

File: code/game/machinery/portable_turret.dm [hunk 1]
```diff
@@ -420,11 +426,12 @@ Status: []<BR>"},
...
 	if (istype(cur_target, /mob/living))
 		var/mob/living/L = cur_target
-		if (L.stat!=2)
-			spawn() popUp()
-			dir=get_dir(src, cur_target)
-			shootAt(cur_target)
-	return
+		if (L.stat==2)
+			return
+	spawn() popUp()
```
Comment (ShiftyRail) [inline code/game/machinery/portable_turret.dm]: I would rather put this under a line break and a tab for style convention.

File: code/game/machinery/portable_turret.dm [hunk 2]
```diff
@@ -420,11 +426,12 @@ Status: []<BR>"},
 /obj/machinery/turret/portable/target()
 	if (istype(cur_target, /mob/living))
 		var/mob/living/L = cur_target
-		if (L.stat!=2)
-			spawn() popUp()
-			dir=get_dir(src, cur_target)
-			shootAt(cur_target)
-	return
+		if (L.stat==2)
```
Comment (ShiftyRail) [inline code/game/machinery/portable_turret.dm]: Define (DEAD)

File: code/game/machinery/turrets.dm [hunk 3]
```diff
@@ -645,6 +650,12 @@
 			if(M.stat || M.lying || (M in exclude))
 				continue
 			pos_targets += M
+		for(var/obj/effect/blob/B in oview(scan_range, src))
+			pos_targets += B
+		for(var/mob/living/simple_animal/hostile/blobspore/BS in oview(scan_range,src))
+			if(BS.stat || BS.lying || (BS in exclude))
```
Comment (ShiftyRail) [inline code/game/machinery/turrets.dm:656]: .gcDestroyed too

**Comment by ShiftyRail** [comment]
What Kurf said, turrets are more to keep constant pressure rather than push against the bleb, unless you emag them and use them long range, which is at your risks and perils

File: code/game/machinery/turrets.dm (hunk 3 above)
Comment (Exxion) [reply code/game/machinery/turrets.dm:656]: It really should not be in `oview()` if it's been `qdel()`ed

**Comment by SonixApache** [comment]
can't wait to see final line of defense clusters of 50 turrets stacked on one tile near cargo

File: code/game/machinery/turrets.dm (hunk 3 above)
Comment (ShiftyRail) [reply code/game/machinery/turrets.dm:656]: true dat

**Comment by ShiftyRail** [comment]
turrets can only fire in one direction, and the blob only needs to expand onto 10 to 20% of the station.

**Pull request merged**